app/api/bucketlist_api.py

```diff
@@ -0,0 +1,67 @@
+from flask import g, request, jsonify, abort
+from app import app, db
+from app.auth.auth import auth
+from flask_login import login_required
+from app.models import Bucketlist
+
+@app.route('/bucketlists/', methods = ['GET'])
+@login_required
+def get_all_bucketlists(self):
+    """gets all the bucket lists"""
+    bucket = Bucketlist.query.filter_by(created_by=g.user_id)
+    if not bucket or bucket.created_by != g.user_id:
+        return jsonify(message='There is No Bucketlist to display!!!!')
+    else:
+        return bucket, 200
+
+@app.route('/bucketlists/', methods = ['POST'])
+@login_required
+def new_bucketlist():
+    """Create a new bucket list"""
+    if not request.json:
+        abort(400)
+    bucketlist = Bucketlist(title=request.get.json('title'), created_by=g.user_id)
+    db.session.add(bucketlist)
+    db.session.commit()
+    return bucketlist, 201
+
+@app.route('/bucketlists/<int:id>', methods = ['GET'])
+@login_required
+def get_bucketlist(id):
+    """Get single bucket list"""
+    if not request.json:
+        abort(400)
```

Comment:
for better UX, you can return a message to the user by making use of custom error handling so the user knows what exactly he/she is doing wrong.